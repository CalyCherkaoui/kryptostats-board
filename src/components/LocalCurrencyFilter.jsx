import React from 'react';
import { PropTypes } from 'prop-types';
import styles from '../styles/Filter.module.css';

const LocalCurrencyFilter = ({ changeLocalCurrency }) => {
  const localCurrencies = [
    { name: 'USD', keyApi: 'usd' },
    { name: 'EUR', keyApi: 'eur' },
    { name: 'JPY', keyApi: 'jpy' },
    { name: 'CAD', keyApi: 'cad' },
    { name: 'BTC', keyApi: 'btc' },
    { name: 'ETH', keyApi: 'eth' },
  ];

  return (
    <div className={`${styles.filter_wrapper} flex_row`}>
      <div className={`${styles.filter_legend} spaced_typography`}>Convert prices to:</div>
      <select
        name="local_currency"
        id="local_currency"
        className={styles.filter_selector}
        onChange={(e) => changeLocalCurrency(e.target.value)}
      >
        {
          localCurrencies.map(
            (currency) => (
              <option
                value={currency.keyApi}
                key={currency.keyApi}
                className={styles.filter_option}
              >
                {currency.name}
              </option>
            ),
          )
        }
      </select>
    </div>
  );
};

LocalCurrencyFilter.propTypes = {
  changeLocalCurrency: PropTypes.func.isRequired,
};

export default LocalCurrencyFilter;
