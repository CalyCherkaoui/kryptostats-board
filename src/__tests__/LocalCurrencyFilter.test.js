// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import LocalCurrencyFilter from '../components/LocalCurrencyFilter';

describe(
  'Renders correctly The navigation bar component',
  () => {
    const testFunction = () => true;

    it(
      'Matches LocalCurrencyFilter snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <LocalCurrencyFilter changeLocalCurrency={testFunction} />
            </Router>
          </Provider>,
        );
        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
