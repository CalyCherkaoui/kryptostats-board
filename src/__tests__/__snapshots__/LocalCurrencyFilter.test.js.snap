// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly The navigation bar component Matches LocalCurrencyFilter snapshot 1`] = `
<div
  className="filter_wrapper flex_row"
>
  <div
    className="filter_legend spaced_typography"
  >
    Convert prices to:
  </div>
  <select
    className="filter_selector"
    id="local_currency"
    name="local_currency"
    onChange={[Function]}
  >
    <option
      className="filter_option"
      value="usd"
    >
      USD
    </option>
    <option
      className="filter_option"
      value="eur"
    >
      EUR
    </option>
    <option
      className="filter_option"
      value="jpy"
    >
      JPY
    </option>
    <option
      className="filter_option"
      value="cad"
    >
      CAD
    </option>
    <option
      className="filter_option"
      value="btc"
    >
      BTC
    </option>
    <option
      className="filter_option"
      value="eth"
    >
      ETH
    </option>
  </select>
</div>
`;
